tools/nim-gdb.py

```diff
@@ -477,9 +477,24 @@ def to_string(self):
   def children(self):
     if self.val:
       length = int(self.val['Sup']['len'])
-      #align = len(str(length - 1))
-      for i in range(length):
-        yield ("data[{0}]".format(i), self.val["data"][i])
+      if length > 0:  
+        inaccessible = False
+        ptr = self.val['data'] if self.val['data'].type.code == gdb.TYPE_CODE_PTR else self.val['data'][0].address
+        if self.val['data'].type.code == gdb.TYPE_CODE_PTR:
+          gdb.write("pointer!\n", gdb.STDLOG)
+        for i in range(length):
+          element_ptr = ptr + i
+          gdb.write("index: " + str(i) + " length: " + str(length) + " sizeof: " + str(self.val['data'][i].type.sizeof) + "\n", gdb.STDLOG)
```

Comment:
We support Python2 ?, because otherwise this can use f-string:

```python
gdb.write(f"index: {i} length: {length} sizeof: {str(self.val['data'][i].type.sizeof)}\n", gdb.STDLOG)
```

If can not be f-string, then can be something like:

```python
gdb.write("index: {0} length: {1} sizeof: {2}\n".format(i, length, self.val['data'][i].type.sizeof), gdb.STDLOG)
```

The chain of `+` is bad because Python string duplicates the string on memory each time.

Reply:
Ah yeah, this is sloppy debug code -- most languages have the extra allocation issue it seems. I've choosen to f the strings, will try and use that elsewhere too.